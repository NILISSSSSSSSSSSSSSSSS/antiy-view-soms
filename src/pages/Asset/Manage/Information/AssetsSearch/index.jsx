import React, { Component } from 'react'
import { connect } from 'dva'
import DateRange from '@/components/common/DateRange'
import {
  Button, Input, Form, Icon,
  Select, TreeSelect, Row
} from 'antd'
import { analysisUrl, subNodeQuery } from '@/utils/common'
import PropTypes from 'prop-types'
import { uniqBy } from 'lodash'
import moment from 'moment'
import { withRouter } from 'dva/router'
import { ASSET_STATUS, ASSETS_IMPORTANT, MAINTENANCE_TYPE,
  ACCESS_STATE, ASSET_SOURCE, IS_BORROW, NET_CONNECTIONS
} from '@a/js/enume'

const FormItem = Form.Item
const { Option } = Select
const { TreeNode, SHOW_PARENT } = TreeSelect
const ASSETS_TYPES = ['计算设备', '网络设备', '安全设备', '存储设备', '其它设备']

@withRouter
@Form.create()
@connect(({ information }) => ({
  getAssetsOS: information.getAssetsOS,
  getManufacturerInfo: information.getManufacturerInfo,
  getGroupInfo: information.getGroupInfo,
  getUserInAsset: information.getUserInAsset,
  getUserAreaTree: information.getUserAreaTree,
  baselineTemplate: information.baselineTemplate
}))
class AssetsSearch extends Component {
  state = {
    isExpand: false  //是否展开
  }

  static propTypes = {
    getAssetsOS: PropTypes.array,  //操作系统
    getManufacturerInfo: PropTypes.array,
    getGroupInfo: PropTypes.array,
    getUserInAsset: PropTypes.array,
    getUserAreaTree: PropTypes.object,
    baselineTemplate: PropTypes.array
  }

  //展开搜索条件
  showCondition = () => {
    this.setState({ isExpand: !this.state.isExpand })
  }

  //加载区域树结构的下拉列表
  getTreeNode = data => {
    if(data) return (
      <TreeNode value={data.stringId} title={data['fullName']} key= {`${data.stringId}`}>
        {data.childrenNode && data.childrenNode.length ? (
          data.childrenNode.map(item =>
            this.getTreeNode(item)
          )
        ) : null
        }
      </TreeNode>
    )
  }

  //执行查询
  handleSubmit = async () => {
    const values = this.getData()
    await this.props.handleSubmit(values)
  }

  //查询时间控制
  submitControl = (val, type) => {
    return moment(moment(val).format('YYYY-MM-DD') + type).unix() * 1000
  }

  // 执行查询
  getData = (e) => {
    let valuesData = {}
    const { getUserAreaTree, form: { validateFields } } = this.props
    // 首次发现时间起始时间 firstEnterStartTime
    // 首次发现时间结束时间 firstEnterEndTime
    // 到期时间起始时间 serviceLifeStartTime
    // 到期时间结束时间 serviceLifeEndTime

    validateFields((err, values = {}) => {
      if (err) return void (0)
      const [FirstBegin, FirstEnd] = values.gmtCreate || [null, null]
      const [ServiceBegin, ServiceEnd] = values.serviceLife || [null, null]
      values.firstEnterStartTime = FirstBegin ? this.submitControl(FirstBegin, ' 00:00:00') : ''
      values.firstEnterEndTime = FirstEnd ? this.submitControl(FirstEnd, ' 23:59:59') : ''
      values.serviceLifeStartTime = ServiceBegin ? this.submitControl(ServiceBegin, ' 00:00:00') : ''
      values.serviceLifeEndTime = ServiceEnd ? this.submitControl(ServiceEnd, ' 23:59:59') : ''
      // values.areaIds = subNodeQuery(getUserAreaTree)(values.areaIds)
      values.isExpand = this.state.isExpand
      valuesData = values
    })
    return valuesData
  }

  //查询条件重置
  handleReset = () => {
    this.resetKeyBegin = 'beigin' + Math.random()
    this.resetKeyEnd = 'end' + Math.random()
    this.props.form.resetFields()
    const values = this.getData()
    this.props.handleSubmit(values, true)
  }

  //**副作用 */
  getManufacturerInfo=(supplier = null)=>{
    this.props.dispatch({ type: 'information/getManufacturerInfo', payload: { supplier } }) // 厂商
  }

  render () {
    const { isExpand } = this.state
    const {
      getAssetsOS,
      getManufacturerInfo,
      getGroupInfo,
      getUserInAsset,
      getUserAreaTree,
      baselineTemplate,
      form: { getFieldDecorator }
    } = this.props

    return (
      <div className="search-bar">
        <Form className="filter-form new-flex-layout" layout="inline">
          <FormItem label="综合查询">
            {getFieldDecorator('multipleQuery')(
              <Input autoComplete='off' className="filter-form-item" maxLength={30} placeholder="名称/编号机器名/国资码/序列号/IP/MAC" />
            )}
          </FormItem>
          <FormItem label='资产类型'>
            {getFieldDecorator('categoryModels')(
              <Select placeholder="全部"
                allowClear
                getPopupContainer={triggerNode => triggerNode.parentNode}
                mode='multiple'>
                {ASSETS_TYPES && ASSETS_TYPES.map((item, i) => (
                  <Option value={i + 1} key={i}>{item}</Option>)
                )}
              </Select>
            )}
          </FormItem>
          <FormItem label='资产状态' className='item-separation'>
            {getFieldDecorator('assetStatusList')(
              <Select placeholder="全部"
                allowClear
                getPopupContainer={triggerNode => triggerNode.parentNode}
                mode='multiple'
              >
                {ASSET_STATUS.map(item => (
                  <Option value={item.value} key={item.value}>{item.name}</Option>)
                )}
              </Select>
            )}
          </FormItem>
          <FormItem className="search-item search-more-item item-separation">
            <Button type="primary" htmlType="submit" className="margin-right" onClick={this.handleSubmit}>查询</Button>
            <Button type="primary" ghost onClick={this.handleReset}>重置</Button>
          </FormItem>
          <FormItem>
            <span className="show-ondition" onClick={this.showCondition}>
              <Icon type={isExpand ? 'up' : 'down'} />
              {isExpand ? '收起' : '高级查询'}
            </span>
          </FormItem>

          <section className="hide-form" style={{ display: isExpand ? 'block' : 'none' }}>
            <Row className='new-flex-layout no-padding'>
              <FormItem label='物理位置'>
                {getFieldDecorator('physicalPosition')(
                  <Input autoComplete='off' placeholder="请输入" maxLength={30} />
                ) }
              </FormItem>
              <FormItem label='机房位置' className='item-separation'>
                {getFieldDecorator('computerRoomLocation')(
                  <Input autoComplete='off' placeholder="请输入"  maxLength={30} />
                ) }
              </FormItem>
              <FormItem label='描述'>
                {getFieldDecorator('describe')(
                  <Input autoComplete='off' placeholder="请输入"  maxLength={30} />
                ) }
              </FormItem>
              <FormItem label='key'>
                {getFieldDecorator('key')(
                  <Input autoComplete='off' placeholder="请输入"  maxLength={30} />
                ) }
              </FormItem>
              <FormItem label='名称' className='item-separation'>
                {getFieldDecorator('name')(
                  <Select showSearch
                    placeholder="全部"
                    getPopupContainer={triggerNode => triggerNode.parentNode}
                    mode='multiple'
                  >
                    <Option value={''} key={''}>全部</Option>
                    {/* {getGroupInfo && getGroupInfo.map((item) => (
                      <Option value={item.id} key={item.id}>{item.value}</Option>)
                    )} */}
                  </Select>
                )}
              </FormItem>
              <FormItem label='厂商'>
                {getFieldDecorator('manufacturer')(
                  <Select onSearch={val=>{this.getManufacturerInfo(val)}}showSearch getPopupContainer={triggerNode => triggerNode.parentNode}>
                    <Option value={''} key={''}>全部</Option>
                    {getManufacturerInfo && getManufacturerInfo.map((item, index) => (
                      <Option value={item} key={index}>{item}</Option>
                    ))}
                  </Select>
                )}
              </FormItem>
              <FormItem label='版本'>
                {getFieldDecorator('Edition', {
                  initialValue: ''
                })(
                  <Select
                    allowClear
                    placeholder="全部"
                    getPopupContainer={triggerNode => triggerNode.parentNode}
                  >
                    <Option value={''} key={''}>全部</Option>
                    {/* {getGroupInfo && getGroupInfo.map((item) => (
                      <Option value={item.id} key={item.id}>{item.value}</Option>)
                    )} */}
                  </Select>
                )}
              </FormItem>
              <FormItem label='操作系统' className='item-separation'>
                {getFieldDecorator('operationSystem', {
                  initialValue: ''
                })(
                  <Select showSearch optionFilterProp="children" getPopupContainer={triggerNode => triggerNode.parentNode}>
                    <Option value={''} key={''}>全部</Option>
                    {getAssetsOS && uniqBy(getAssetsOS, 'value').map((item) => (
                      <Option value={item.id} key={item.id}>{item.value}</Option>
                    ))}
                  </Select>
                )}
              </FormItem>
              <FormItem label='资产来源'>
                {getFieldDecorator('assetSource')(
                  <Select placeholder="全部"
                    getPopupContainer={triggerNode => triggerNode.parentNode}
                    showSearch={false}>
                    <Option value={''} key={''}>全部</Option>
                    {ASSET_SOURCE.map((item, i) => (
                      <Option value={i + 1} key={i}>{item}</Option>)
                    )}
                  </Select>
                )}
              </FormItem>
              <FormItem label='从属业务'>
                {getFieldDecorator('assetSource666')(
                  <Select placeholder="全部"
                    getPopupContainer={triggerNode => triggerNode.parentNode}
                    showSearch={false}>
                    <Option value={''} key={''}>全部</Option>
                  </Select>
                )}
              </FormItem>
              <FormItem label='资产组' className='item-separation'>
                {getFieldDecorator('assetGroup', {
                  initialValue: ''
                })(
                  <Select showSearch
                    getPopupContainer={triggerNode => triggerNode.parentNode}
                    optionFilterProp="children">
                    <Option value={''} key={''}>全部</Option>
                    {getGroupInfo && getGroupInfo.map((item) => (
                      <Option value={item.id} key={item.id}>{item.value}</Option>)
                    )}
                  </Select>
                )}
              </FormItem>
              <FormItem label='网络类型'>
                {getFieldDecorator('Networktype', {
                  initialValue: ''
                })(
                  <Select
                    allowClear
                    placeholder="全部"
                    getPopupContainer={triggerNode => triggerNode.parentNode}
                  >
                    <Option value={''} key={''}>全部</Option>
                    {/* {getGroupInfo && getGroupInfo.map((item) => (
                      <Option value={item.id} key={item.id}>{item.value}</Option>)
                    )} */}
                  </Select>
                )}
              </FormItem>
              <FormItem label='归属区域'>
                {getFieldDecorator('areaIds')(
                  <TreeSelect
                    showSearch
                    dropdownStyle={{ maxHeight: 400, overflow: 'auto' }}
                    placeholder="全部"
                    allowClear
                    multiple
                    getPopupContainer={triggerNode => triggerNode.parentNode}
                    treeDefaultExpandAll
                    // showCheckedStrategy={SHOW_ALL}
                    // treeCheckable={true}
                    treeNodeFilterProp='title'
                  >
                    {this.getTreeNode(getUserAreaTree)}
                  </TreeSelect>
                )}
              </FormItem>
              <FormItem label='使用者' className='item-separation'>
                {getFieldDecorator('responsibleUserId', {
                  initialValue: ''
                })(
                  <Select showSearch optionFilterProp="children" getPopupContainer={triggerNode => triggerNode.parentNode}>
                    <Option value={''} key={''}>全部</Option>
                    {getUserInAsset && getUserInAsset.map((item) => (
                      <Option value={item.id} key={item.id}>{item.value}</Option>
                    ))}
                  </Select>
                )}
              </FormItem>
              <FormItem label='准入状态'>
                {getFieldDecorator('AccessState', {
                  initialValue: ''
                })(
                  <Select
                    allowClear
                    placeholder="全部"
                    getPopupContainer={triggerNode => triggerNode.parentNode}
                  >
                    <Option value={''} key={''}>全部</Option>
                    {ACCESS_STATE.map((item) => (
                      <Option value={item.value} key={item.value}>{item.value}</Option>)
                    )}
                  </Select>
                )}
              </FormItem>
              <FormItem label='是否可借用'>
                {getFieldDecorator('WhetherBorrowed', {
                  initialValue: ''
                })(
                  <Select
                    allowClear
                    placeholder="全部"
                    getPopupContainer={triggerNode => triggerNode.parentNode}
                  >
                    <Option value={''} key={''}>全部</Option>
                    {IS_BORROW.map((item) => (
                      <Option value={item.value} key={item.value}>{item.value}</Option>)
                    )}
                  </Select>
                )}
              </FormItem>
              <FormItem label='基准模板' className='item-separation'>
                {getFieldDecorator('baselineTemplateId', {
                  initialValue: ''
                })(
                  <Select showSearch optionFilterProp="children" getPopupContainer={triggerNode => triggerNode.parentNode}>
                    <Option value={''} key={''}>全部</Option>
                    {baselineTemplate && baselineTemplate.map((item) => (
                      <Option value={item.id} key={item.id}>{item.value}</Option>
                    ))}
                  </Select>
                )}
              </FormItem>
              <FormItem label='网络连接'>
                {getFieldDecorator('netConnonet', {
                  initialValue: ''
                })(
                  <Select allowClear
                    placeholder="全部"
                    getPopupContainer={triggerNode => triggerNode.parentNode}>
                    <Option value={''} key={''}>全部</Option>
                    {NET_CONNECTIONS.map((item) => (
                      <Option value={item.value} key={item.value}>{item.value}</Option>)
                    )}
                  </Select>
                )}
              </FormItem>
              <FormItem label='是否涉密'>
                {getFieldDecorator('IsConfidential', {
                  initialValue: ''
                })(
                  <Select
                    allowClear
                    placeholder="全部"
                    getPopupContainer={triggerNode => triggerNode.parentNode}
                  >
                    <Option value={''} key={''}>全部</Option>
                    {IS_BORROW.map((item) => (
                      <Option value={item.value} key={item.value}>{item.value}</Option>)
                    )}
                  </Select>
                )}
              </FormItem>
              <FormItem label='是否孤岛设备' className='item-separation'>
                {getFieldDecorator('IslandEquipment', {
                  initialValue: ''
                })(
                  <Select
                    allowClear
                    placeholder="全部"
                    getPopupContainer={triggerNode => triggerNode.parentNode}
                  >
                    <Option value={''} key={''}>全部</Option>
                    {IS_BORROW.map((item) => (
                      <Option value={item.value} key={item.value}>{item.value}</Option>)
                    )}
                  </Select>
                )}
              </FormItem>
              <FormItem label='重要程度'>
                {getFieldDecorator('importanceDegree')(
                  <Select placeholder="全部"
                    getPopupContainer={triggerNode => triggerNode.parentNode}
                    showSearch={false}>
                    <Option value={''} key={''}>全部</Option>
                    {ASSETS_IMPORTANT.map((item, i) => (
                      <Option value={item.vaule} key={i}>{item.name}</Option>)
                    )}
                  </Select>
                )}
              </FormItem>
              <FormItem label='维护方式'>
                {getFieldDecorator('Maintenance')(
                  <Select placeholder="全部"
                    getPopupContainer={triggerNode => triggerNode.parentNode}
                    showSearch={false}>
                    <Option value={''} key={''}>全部</Option>
                    {MAINTENANCE_TYPE.map((item, i) => (
                      <Option value={item.vaule} key={i}>{item.name}</Option>)
                    )}
                  </Select>
                )}
              </FormItem>

              <FormItem label="到期时间" className='item-date-container item-separation'>
                {getFieldDecorator('serviceLife', {
                })(
                  <DateRange future placeholder={['开始时间', '结束时间']} format="YYYY-MM-DD" resetKey={this.resetKeyBegin} />
                )}
              </FormItem>
              <FormItem label="首次发现时间" className='item-date-container'>
                {getFieldDecorator('gmtCreate', {
                })(
                  <DateRange future={false} placeholder={['开始时间', '结束时间']} format="YYYY-MM-DD" resetKey={this.resetKeyBegin} />
                )}
              </FormItem>

              <FormItem label="启用时间" className='item-date-container'>
                {getFieldDecorator('serviceLife1', {
                })(
                  <DateRange future placeholder={['开始时间', '结束时间']} format="YYYY-MM-DD" resetKey={this.resetKeyBegin} />
                )}
              </FormItem>

              <FormItem label="装机时间" className='item-date-container item-separation'>
                {getFieldDecorator('serviceLife2', {
                })(
                  <DateRange future placeholder={['开始时间', '结束时间']} format="YYYY-MM-DD" resetKey={this.resetKeyBegin} />
                )}
              </FormItem>
            </Row>
          </section>
        </Form>
      </div>

    )
  }

  componentDidMount () {
    const { dispatch, form: { setFieldsValue } } = this.props
    const init = analysisUrl(this.props.location.search)
    if(init.conditionShow) this.setState({ isExpand: init.conditionShow })
    if(init.assetStatusList) setFieldsValue({ assetStatusList: [Number(init.assetStatusList)] })
    dispatch({ type: 'information/getAssetsOS' }) // 操作系统
    dispatch({ type: 'information/getGroupInfo' }) // 资产组
    this.getManufacturerInfo() // 厂商
    dispatch({ type: 'information/getUserInAsset' }) // 使用者
    dispatch({ type: 'information/getUserAreaTree' }) // 区域树
    dispatch({ type: 'information/baselineTemplate' }) //  基准模板
    // this.handleSubmit()
    // this.props.handleSubmit(values)
  }

  UNSAFE_componentWillReceiveProps (nextProps) {
    // 更新列表数据
    if (JSON.stringify(this.props.isExpand) !== JSON.stringify(nextProps.isExpand)) {
      this.setState({
        isExpand: nextProps.isExpand
      })
    }
  }

}

export {
  AssetsSearch,
  ASSETS_TYPES
}
