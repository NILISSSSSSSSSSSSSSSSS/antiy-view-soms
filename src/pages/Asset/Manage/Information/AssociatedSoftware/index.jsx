import React, { PureComponent } from 'react'
import { connect } from 'dva'
import { Search, CommonModal } from '@c'
import { Form, Table, message, Select, Input } from 'antd'
import PropTypes from 'prop-types'
import { debounce, throttle } from 'lodash'
import { TooltipFn } from '@u/common'
import api from '@/services/api'

const FormItem = Form.Item
const { Option } = Select
@Form.create()
@connect(({ asset }) => ({
  installableList: asset.installableList,
  usersByRoleCodeAndAreaIdList: asset.usersByRoleCodeAndAreaIdList
}))
class AssociatedSoftware extends PureComponent{
   state = {
     PrefixCls: 'AssociatedSoftware',
     //查询开始
     currentPage: 1,
     pageSize: 10,
     selectedRowKeys: [],
     supplier: null,
     productName: null
     //查询结束
   }

    static propTypes = {
      soVisible: PropTypes.bool,
      associds: PropTypes.array,
      associatedSoftware: PropTypes.func,
      installableList: PropTypes.object,
      usersByRoleCodeAndAreaIdList: PropTypes.array
    }

    onSelectChange = (selectedRowKeys, selectedRows) => {
      this.setState({ otherBusinessIds: selectedRowKeys, selectedRowKeys })
    };

    //提交
    onSubmit = (values) => {
      this.setState({
        supplier: values.supplier,
        productName: values.productName,
        currentPage: 1,
        pageSize: 10
      }, this.installableList)
    }

    //重置
    onReset = ()=>{
      this.setState({
        supplier: null,
        productName: null,
        currentPage: 1,
        pageSize: 10
      }, this.installableList)
    }

    //分页
    pageChange=(currentPage, pageSize)=>{
      this.setState({
        currentPage,
        pageSize
        // selectedRowKeys: []
      }, this.installableList)
    }

    getRecordsArrIds=(stringId)=>{
      let assetId = []
      this.props.associds.forEach(v=>assetId.push(v[stringId]))
      return assetId
    }

    //确定关联组件
    onConfirm = throttle(()=>{
      this.props.form.validateFields((err, values) => {
        if(err) return void(0)
        this.onConfirmCB(values)
      })
    }, 5000, { trailing: false })

    /***************副作用开始************** */

    onConfirmCB=values=>{
      const {
        associatedSoftware,
        usersByRoleCodeAndAreaIdList,
        compEffectCB
      } = this.props
      let params = {
        assetId: this.getRecordsArrIds('stringId'),
        softId: this.state.selectedRowKeys
      }
      if(params.softId.length === 0){
        message.info('请选择关联软件!')
        return void(0)
      }
      if(values.baselineConfigUserId === ' '){
        values.baselineConfigUserId = usersByRoleCodeAndAreaIdList.map(item=>item.stringId).join(',')
      }
      params.manualStartActivityRequest = {
        formData: {
          'baselineConfigUserId': values.baselineConfigUserId,
          'memo': values.memo
        }
      }
      api.batchRelation(params).then(res => {
        if(res && res.head && res.head.code === '200'){
          message.success('操作成功！')
          associatedSoftware(false)
          compEffectCB()
        }
      })
    }

    installableList=debounce(async ()=>{
      const state = this.state
      const payload = {
        baselineTemplateId: this.props.associds[0].baselineTemplateId,
        supplier: state.supplier,
        productName: state.productName,
        currentPage: state.currentPage,
        pageSize: state.pageSize
      }
      await this.props.dispatch({ type: 'asset/installableList', payload: payload })
    }, 300)

    //操作人员
    getUsersByRoleCodeAndAreaId=()=>{
      this.props.dispatch({ type: 'asset/getUsersByRoleCodeAndAreaId', payload: {
        flowNodeTag: 'config_base', flowId: 4, areaId: this.getRecordsArrIds('areaId') }
      })
    }

    render (){
      const { selectedRowKeys, currentPage, pageSize } = this.state
      const { soVisible, associatedSoftware, installableList, form: { getFieldDecorator }, usersByRoleCodeAndAreaIdList } = this.props
      const defaultFields = [
        { type: 'input', label: '厂商', placeholder: '请输入厂商', key: 'supplier', allowClear: true, maxLength: 30 },
        { type: 'input', label: '名称', placeholder: '请输入名称', key: 'productName', allowClear: true, maxLength: 30 }
      ]

      const columns = [
        {
          title: '厂商',
          dataIndex: 'supplier',
          render: text => TooltipFn (text)
        },
        {
          title: '名称',
          dataIndex: 'productName',
          render: text => TooltipFn (text)
        },
        {
          title: '版本',
          dataIndex: 'version',
          render: text => TooltipFn (text)
        },
        {
          title: '软件平台',
          dataIndex: 'softPlatform',
          render: text => TooltipFn (text)
        }
      ]

      const rowSelection = {
        selectedRowKeys,
        onChange: this.onSelectChange
      }

      const formLayout = {
        labelCol: {
          span: 8
        },
        wrapperCol: {
          span: 16
        }
      }

      return(
        <CommonModal
          title="关联软件"
          type="search"
          visible={soVisible}
          width={1200}
          oktext='保存'
          onConfirm={this.onConfirm}
          onClose={()=>associatedSoftware(false)}>
          <div className="table-wrap">
            <Search defaultFields={defaultFields} onSubmit={this.onSubmit} onReset={this.onReset}/>
            <Table
              rowKey='softwareId'
              columns={columns}
              rowSelection={rowSelection}
              dataSource={installableList.items || []}
              pagination={{
                current: currentPage,
                pageSize: pageSize,
                onShowSizeChange: this.pageChange,
                onChange: this.pageChange,
                showQuickJumper: true,
                showSizeChanger: installableList.totalRecords > 10,
                pageSizeOptions: ['10', '20', '30', '40'],
                showTotal: () => `共 ${installableList.totalRecords || 0} 条数据`,
                total: installableList.totalRecords || 0
              }}
            />
            <Form className="filter-form new-flex-layout" layout="inline" onSubmit={this.handleSubmit}>
              <FormItem {...formLayout} label='下一步骤'>
                {getFieldDecorator('666')(
                  <span>基准配置</span>
                ) }
              </FormItem>
              <FormItem {...formLayout} label='下一步执行人'>
                {getFieldDecorator('baselineConfigUserId', {
                  initialValue: ' ',
                  rules: [{ required: true, message: '请指派人员！' }]
                })(
                  <Select showSearch
                    allowClear
                    getPopupContainer={triggerNode => triggerNode.parentNode}
                    optionFilterProp="children">
                    {usersByRoleCodeAndAreaIdList.length ? <Option value=' '>全部</Option> : ''}
                    {usersByRoleCodeAndAreaIdList.map(
                      (now, index)=>(<Option value = {now.stringId} key={index + 'baselineConfigUserId'}>{now.name}</Option>))
                    }
                  </Select>
                ) }
              </FormItem>
              <FormItem {...formLayout} label='备注'>
                {getFieldDecorator('memo', {
                  rules: [
                    { message: '最多300个字符！', max: 300 }
                  ]
                })(
                  <Input autoComplete='off' className="filter-form-item" placeholder="请输入" />
                ) }
              </FormItem>
            </Form>
          </div>
        </CommonModal>
      )
    }

    async componentDidMount (){
      await Promise.all([
        this.installableList(),
        this.getUsersByRoleCodeAndAreaId()
      ])
    }

    componentWillUnmount (){
      this.props.dispatch({ type: 'asset/save', payload: { installableList: {} } })
    }
}

export default AssociatedSoftware
