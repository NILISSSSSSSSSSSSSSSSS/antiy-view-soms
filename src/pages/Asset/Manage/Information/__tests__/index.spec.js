import React from 'react'
import Information  from '../index'
import { shallow } from 'enzyme'
import PropTypes from 'prop-types'
import api from '../../../../../services/api'
// import hasAuth from '@/u/auth'

jest.mock('lodash', () => ({
  debounce: jest.fn((fn => fn)),
  cloneDeep: jest.fn((fn => fn))
}))
jest.mock('dva/router')
jest.mock('@/a/permission')
jest.mock('@/u/auth', ()=>({
  hasAuth: jest.fn((fn => true))
}))

// hasAuth = jest.fn((fn => true))

jest.mock('../../../../../services/api')

const props = {
  location: {
    search: null
  },
  history: {
    push: jest.fn()
  }
}
const assetBody = {
  items: [
    {
      disabled: true,
      stringId: 666,
      name: 666,
      number: 666,
      categoryModel: 666,
      ips: 666,
      macs: 666,
      assetStatus: 6,
      gmtCreate: 666,
      serviceLife: 666,
      assetGroup: 666,
      operationSystemName: 666,
      assetSource: 666,
      importanceDegree: 666,
      manufacturer: 666,
      baselineTemplateName: 666,
      areaName: 666,
      responsibleUserName: 666,
      waitingTaskReponse: {
        taskId: 666
      }
    }
  ],
  totalRecords: 20
}
const store = {
  subscribe: () => ({
  }),
  dispatch: () =>({}),
  getState: () => ({
    asset: {
      assetBody
    }
  })
}
const options = {
  context: { store },
  childContextTypes: { store: PropTypes.object.isRequired }
}

describe('<Implementation />', () => {

  let wrapper, instance
  const assetBody2 = JSON.parse(JSON.stringify(assetBody))
  assetBody2.items[0].name = 777
  const nextProps = {
    assetBody: assetBody2
  }
  const selectedRowKeys = []
  const selectedRows0 = []
  const selectedRows1 = [{
    assetStatus: 6
  }]
  const selectedRows2 = [{
    baselineTemplateId: 1,
    assetStatus: 6
  }, {
    baselineTemplateId: 2,
    assetStatus: 6
  }]

  beforeEach(() => {
    const assetBody2 = JSON.parse(JSON.stringify(assetBody))
    assetBody2.items[0].name = 777
    const nextProps = {
      assetBody: assetBody2
    }
    wrapper = shallow(<Information {...props}/>).dive(options).dive(options)
    instance = wrapper.instance()
    instance.UNSAFE_componentWillReceiveProps(nextProps)
    // console.log(instance)
  })

  it('render <Information /> component', () => {
    expect(wrapper.find('.table-wrap')).toHaveLength(1)
  })

  it('testFunc UNSAFE_componentWillReceiveProps', () => {
    instance.UNSAFE_componentWillReceiveProps(nextProps)
  })

  it('testFunc tableHeaderChange', () => {
    const column = {
      key: 'baselineTemplateName'
    }
    instance.tableHeaderChange(column)
  })

  it('testFunc onSelectBtn', () => {
    const Arr = [1, 3, 4, 5, 6, 0, 2]
    Arr.forEach(item=>instance.onSelectBtn(item))
  })

  it('testFunc onSelectChange', () => {
    instance.onSelectChange(selectedRowKeys, selectedRows0)
    instance.onSelectChange(selectedRowKeys, selectedRows1)
    instance.onSelectChange(selectedRowKeys, selectedRows2)

    const nextProps2 = JSON.parse(JSON.stringify(nextProps))
    nextProps2.assetBody.items = null
    instance.UNSAFE_componentWillReceiveProps(nextProps)
    instance.onSelectChange(selectedRowKeys, selectedRows2)
  })

  it('testFunc registration', () => {
    instance.registration()
  })

  it('testFunc noRegistration', () => {
    instance.noRegistration(assetBody.items)
    instance.noRegistration(null, true)
  })

  it('testFunc noRegistrationCB', () => {
    api.assetNoRegister = jest.fn()
    api.assetNoRegister.mockReturnValue(
      Promise.resolve({
        head: { code: '200' },
        body: {  }
      })
    )
    instance.noRegistrationCB()
  })

  it('testFunc associatedSoftware', () => {
    instance.associatedSoftware()
    instance.onSelectChange(selectedRowKeys, selectedRows2)
    instance.associatedSoftware()
  })

  it('testFunc nowCLickGetInit', () => {
    const Arr = [
      {
        waitingTaskReponse: { taskId: 666 }
      }
    ]
    instance.nowCLickGetInit(Arr)
  })

  it('testFunc switchCB', () => {
    instance.switchCB(assetBody.items)
  })

  it('testFunc processOnClose', () => {
    instance.processOnClose(assetBody.items)
    instance.processOnClose(assetBody.items, true)
    const Arr = JSON.parse(JSON.stringify(assetBody.items))
    Arr[0].assetStatus = 7
    instance.processOnClose(Arr)
  })

  it('testFunc onRef', () => {
    instance.onRef(assetBody.items, 3)
  })

  it('testFunc pageChange', () => {
    instance.pageChange(1, 10)
  })

  it('testFunc handleTableSort', () => {
    instance.handleTableSort(null, null, { order: 'descend' })
    instance.handleTableSort(null, null, { order: 'ascend' })
    instance.handleTableSort(null, null, { order: null })
  })

  it('testFunc handleSubmit', () => {
    const valuesData = {
      multipleQuery: 666,
      categoryModels: 666,
      assetStatusList: 666,
      assetGroup: 666,
      operationSystem: 666,
      assetSource: 666,
      importanceDegree: 666,
      manufacturer: 666,
      areaIds: 666,
      responsibleUserId: 666,
      baselineTemplateId: 666,
      firstEnterStartTime: 666,
      firstEnterEndTime: 666,
      serviceLifeStartTime: 666,
      serviceLifeEndTime: 666,
      sortName: 666,
      sortOrder: 666,
      currentPage: 1,
      pageSize: 10
    }
    instance.handleSubmit(null)
    instance.handleSubmit(valuesData)

  })

  it('testFunc exportGetData', () => {
    instance.exportGetData()
  })

  it('testFunc operateJump', () => {
    const Arr = [1, 2, 3]
    Arr.forEach(v=> instance.operateJump(v, assetBody.items))
  })

  it('testFunc isTaskIdShow', () => {
    instance.isTaskIdShow(assetBody.items[0])
  })

  it('testFunc operateFunc', () => {
    instance.operateFunc(assetBody.items)
  })

  it('testFunc templateDownload and changeTemplateChecked', () => {
    instance.templateDownload()
    instance.changeTemplateChecked([666])
    instance.templateDownload()
  })

  it('testFunc itemsInit 0', () => {
    api.getAssetList = jest.fn()
    api.getAssetList.mockReturnValue(
      Promise.resolve({
        head: { code: '200' },
        body: { totalRecords: 0 }
      })
    )
    instance.itemsInit(nextProps)
  })

  it('testFunc itemsInit 1', () => {
    api.getAssetList = jest.fn()
    api.getAssetList.mockReturnValue(
      Promise.resolve({
        head: { code: '200' },
        body: { totalRecords: 1 }
      })
    )
    instance.itemsInit(nextProps)
  })

  it('testFunc exportAssets', () => {
    instance.exportAssets()
  })

  it('testFunc nowCLickCb', () => {
    api.getAssetList = jest.fn()
    api.getAssetList.mockReturnValue(
      Promise.resolve({
        head: { code: '201' },
        body: { }
      })
    )
    api.getAssetList()
  })

  it('shoulder render a Implementation',  ()=> {
    const app = wrapper.debug()
    expect(app).toMatchSnapshot()
  })

})