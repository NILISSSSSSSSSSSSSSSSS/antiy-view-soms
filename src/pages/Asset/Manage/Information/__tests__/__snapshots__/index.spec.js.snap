// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Implementation /> shoulder render a Implementation 1`] = `
"<div className=\\"Information\\">
  <Form(Connect(AssetsSearch)) handleSubmit={[Function]} />
  <div className=\\"table-wrap\\">
    <TableBtns leftBtns={{...}} rightBtns={{...}} />
    <Table scroll={{...}} rowKey=\\"stringId\\" className=\\"asset-table \\" rowClassName=\\"\\" onChange={[Function]} columns={{...}} rowSelection={{...}} dataSource={{...}} pagination={{...}} useFixedHeader={false} size=\\"default\\" loading={false} bordered={false} indentSize={20} locale={{...}} showHeader={true} sortDirections={{...}} childrenColumnName=\\"children\\" />
  </div>
  <Form(Connect(Implementation)) onRef={[Function]} getAssetList={[Function]} />
  <Form(Connect(Verification)) onRef={[Function]} getAssetList={[Function]} />
  <Form(Connect(NetworkAccess)) onRef={[Function]} getAssetList={[Function]} />
  <Form(Connect(Retirement)) onRef={[Function]} getAssetList={[Function]} />
  <Form(Connect(Examination)) onRef={[Function]} getAssetList={[Function]} />
  <Form(Connect(Rectification)) onRef={[Function]} getAssetList={[Function]} />
  <Form(Connect(Decommission)) onRef={[Function]} getAssetList={[Function]} />
  <Connect(AssetModal) data={{...}} />
  <Connect(AssetModal) data={{...}} />
</div>"
`;
